File: searching/linear_search/linear_search.go
package main

import "fmt"

func linearSearch(arr []int, target int) {
	for i := 0; i < len(arr); i++ {
		if target != arr[i] {
			continue
		} else {
			fmt.Printf("Target:%d\nFound At Index: %d\n", target, i)
		}
	}
}

func main() {
	var arr = []int{1, 2, 3, 4, 5, 7, 8, 9}
	var target = 7
	linearSearch(arr, target)
}

File: searching/linear_search/linear_search.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
/*
searches for a targert through a sorted array element by element
*/

void linearSearch(vector<int> arr, int target) {
    int flag = 0;
    int index;
    for (int i = 0; i<arr.size(); ++i) {
        if (arr[i] == target) {
            flag = 1;
            index = i;
            break;
        }
    }

    if (flag == 1) {
        cout << "Target Found at Index " << index << "\n";
    } else {
        cout << "Target not found\n";
    }
}

int main() {
    vector<int> arr = {5,7,2,3,8,9,1};
    
    // sort array
    auto first = arr.begin();
    auto last = arr.end();

    sort(first, last);

    cout << "Sorted Array: ";
    for (int i = 0; i<arr.size(); ++i) {
        cout << arr[i];
    }
    cout << "\n";

    linearSearch(arr, 5);
    linearSearch(arr, 55);
    return 0;
}